### PR TITLE
Fix: Add max retries for all jobs

### PR DESCRIPTION
**Pull request by tejaswinichile**

### CODE REVIEW

File: app/jobs/inboxes/fetch_imap_emails_job.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@ class Inboxes::FetchImapEmailsJob < ApplicationJob
   queue_as :low
 
   def perform(channel)
+    return unless channel.imap_enabled?
```
Comment (aswindevps) [inline app/jobs/inboxes/fetch_imap_emails_job.rb:5]: please remove '?'

`return unless channel.imap_enabled`

Comment (tejaswinichile) [reply app/jobs/inboxes/fetch_imap_emails_job.rb:5]: @aswindevps `channel.imap_enabled?` also works as `channel.imap_enabled` it's a ruby convention to write it with a question mark at the end if it's returning a boolean value.

Comment (sojan-official) [reply app/jobs/inboxes/fetch_imap_emails_job.rb:5]: this should be okay. ref: https://stackoverflow.com/a/1345854/939299

**Comment by sojan-official** [comment]
@tejaswinichile  seems like we will have to bump up to rails 6.1.4.2

 https://weblog.rubyonrails.org/2021/12/14/Rails-6-0-4-2-6-1-4-2-and-7-0-0-rc2-have-been-released/

**Comment by tejaswinichile** [comment]
@sojan-official I added it in the different PR, let me add it here.

**Review by sojan-official — APPROVED** [review]

**Pull request merged**